aws/rust-runtime/aws-config/src/default_provider/disable_request_compression.rs

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::environment::parse_bool;
+use crate::provider_config::ProviderConfig;
+use aws_runtime::env_config::EnvConfigValue;
+use aws_smithy_types::error::display::DisplayErrorContext;
+
+mod env {
+    pub(super) const DISABLE_REQUEST_COMPRESSION: &str = "AWS_DISABLE_REQUEST_COMPRESSION";
+}
+
+mod profile_key {
+    pub(super) const DISABLE_REQUEST_COMPRESSION: &str = "disable_request_compression";
+}
+
+/// Load the value for "disable request compression".
+///
+/// This checks the following sources:
+/// 1. The environment variable `AWS_DISABLE_REQUEST_COMPRESSION=true/false`
+/// 2. The profile key `disable_request_compression=true/false`
+///
+/// If invalid values are found, the provider will return None and an error will be logged.
+pub async fn disable_request_compression_provider(
```

Comment:
Seems odd to me these end up being public to the world but that seems to be the pattern here. I would have thought `pub crate`.

Reply:
I was doing it to match the other providers but I don't mind making them `pub(crate)`. We can always undo that later if we decide to.